.github/workflows/flutter_integration_test.yml

```diff
@@ -46,11 +46,39 @@ jobs:
       - name: flutter pub get
         run: flutter pub get
 
+      - name: Gradle cache
+        uses: gradle/gradle-build-action@v2
+
+      - name: AVD cache
+        uses: actions/cache@v3
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-21
+
+      - name: create AVD and generate snapshot for caching
+        if: steps.avd-cache.outputs.cache-hit != 'true'
+        uses: reactivecircus/android-emulator-runner@d94c3fbe4fe6a29e4a5ba47c12fb47677c73656b #pin@v2.28.0
+        with:
+          working-directory: ./flutter/example
+          api-level: 21
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: false
+          arch: x86_64
+          profile: Nexus 6
+          script: echo "Generated AVD snapshot for caching."
+
       - name: launch android emulator & run android integration test
-        uses: reactivecircus/android-emulator-runner@50986b1464923454c95e261820bc626f38490ec0 #pin@v2.27.0
+        uses: reactivecircus/android-emulator-runner@d94c3fbe4fe6a29e4a5ba47c12fb47677c73656b #pin@v2.28.0
         with:
           working-directory: ./flutter/example
           api-level: 21
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
```

Comment:
Here it's not that long, so it's good, but in case there would be more params we can use YML multiline string `>` that replaces new lines with spaces.

```yml
          emulator-options: >
            -noaudio
            ...
```

https://github.com/getsentry/sentry-react-native/blob/da1b6a88c430816aafb4ed733863395007d667c8/.github/workflows/e2e.yml#L434-L435